lib/ansible/module_utils/docker_common.py

```diff
@@ -410,12 +411,20 @@ def _image_lookup(self, name, tag):
         exists.
         '''
         try:
-            response = self.images(name=name)
+            if name in DOCKER_RESERVED_IMAGE_NAMES:
+                response = self.images()
+            else:
+                response = self.images(name=name)
         except Exception as exc:
             self.fail("Error searching for image %s - %s" % (name, str(exc)))
         images = response
-        if tag:
-            lookup = "%s:%s" % (name, tag)
+        lookup = "%s:%s" % (name, tag)
+        if tag and name in DOCKER_RESERVED_IMAGE_NAMES:
+            for image in response:
+                if image['Id'] == lookup:
```

Comment:
Is `image[Id]` a long `sha256` hash like `d3d92961e7c244734d6bb307dfecab651e96d4b6e18f4390f734cca477453302`? 
Should we allow users to specify it with a short one like `git`?.